common/os.go

```diff
@@ -93,28 +93,30 @@ func MustWriteFile(filePath string, contents []byte, mode os.FileMode) {
 	}
 }
 
-// Writes to newBytes to filePath.
-// Guaranteed not to lose *both* oldBytes and newBytes,
-// (assuming that the OS is perfect)
+// WriteFileAtomic writes newBytes to temp and atomically moves to filePath
+// when everything else succeeds.
 func WriteFileAtomic(filePath string, newBytes []byte, mode os.FileMode) error {
-	// If a file already exists there, copy to filePath+".bak" (overwrite anything)
-	if _, err := os.Stat(filePath); !os.IsNotExist(err) {
-		fileBytes, err := ioutil.ReadFile(filePath)
-		if err != nil {
-			return fmt.Errorf("Could not read file %v. %v", filePath, err)
-		}
-		err = ioutil.WriteFile(filePath+".bak", fileBytes, mode)
-		if err != nil {
-			return fmt.Errorf("Could not write file %v. %v", filePath+".bak", err)
-		}
+	f, err := ioutil.TempFile("", "")
```

Comment:
>  I suppose its not a real issue

Because we could crash right before we write to the temp file too. So we'd be marginally better off if we knew where the temp file was, but not much.